src/arena/cfr/export.rs

```diff
@@ -0,0 +1,488 @@
+//! CFR State Tree Visualization
+//! 
+//! This module provides functionality to visualize the Counterfactual Regret Minimization (CFR)
+//! state tree in various formats. The visualization helps in understanding and debugging the
+//! poker game tree structure and decision points.
+//!
+//! # Features
+//!
+//! - Multiple export formats:
+//!   - DOT (Graphviz format)
+//!   - PNG (static image)
+//!   - SVG (scalable, interactive in browsers)
+//!
+//! - Node visualization:
+//!   - Root nodes: Light blue, double octagon shape
+//!   - Chance nodes: Light green, ellipse shape - shows possible card deals
+//!   - Player nodes: Coral color, box shape - shows player seat and action choices
+//!   - Terminal nodes: Light grey, hexagon shape - shows utility values
+//!
+//! - Edge information:
+//!   - For chance nodes: Labels show which cards could be dealt
+//!   - For player nodes: Labels show actions (fold at index 0)
+//!   - Edge thickness indicates frequency of use
+//!
+//! ```
+
+use std::fs::File;
+use std::io::{self, Write};
+use std::path::Path;
+
+use crate::core::Card;
+
+use super::{CFRState, NodeData};
+
+/// Exports the CFR state tree to a DOT (Graphviz) file.
+/// 
+/// This function creates a DOT file representation of the CFR state tree
+/// which can be visualized using Graphviz tools or online viewers.
+/// 
+/// # Arguments
+/// 
+/// * `state` - A reference to the CFR state to export
+/// * `output_path` - The path where the DOT file will be saved
+/// 
+/// # Returns
+/// 
+/// * `io::Result<()>` - Success or error
+pub fn export_to_dot(state: &CFRState, output_path: &Path) -> io::Result<()> {
+    let mut file = File::create(output_path)?;
```

Comment:
Rather than always writing to a file, can we create the dot in memory in one function and then write it to the file in another? This would allow testing that the export is producing the correct dot output.